src/main/java/seedu/address/model/leave/UniqueLeaveList.java

```diff
@@ -0,0 +1,150 @@
+package seedu.address.model.leave;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.leave.exceptions.DuplicateLeaveException;
+import seedu.address.model.leave.exceptions.LeaveNotFoundException;
+
+/**
+ * A list of leaves that enforces uniqueness between its elements and does not allow nulls.
+ * A leave is considered unique by comparing using {@code Leave#isSameLeave(Leave)}. As such, adding and updating of
+ * leaves uses Leave#isSameLeave(Leave) for equality so as to ensure that the leave being added or updated is
+ * unique in terms of identity in the UniqueLeaveList. However, the removal of a leave uses Leave#equals(Object) so
+ * as to ensure that the leave with exactly the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Leave#isSameLeave(Leave)
+ */
+public class UniqueLeaveList implements Iterable<Leave> {
+
+    private final ObservableList<Leave> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Leave> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent leave as the given argument.
+     */
+    public boolean contains(Leave toCheck) {
```

Comment:
Might be a bit of a nitpick but would it be better to change the parameter name to a noun e.g. target?

Reply:
Noted, will change this accordingly, thank you